configs/opencv-ovms/cmd_client/main.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+)
+
+type OvmsClientInfo struct {
+	PipelineScript    string
+	PipelineInputArgs string
+}
+type OvmsClientConfig struct {
+	OvmsClient OvmsClientInfo
+}
+
+func main() {
+	// load config yaml from file
+	log.Println("Loading configuration yaml file from ./res folder...")
+	contents, err := os.ReadFile("./res/configuration.yaml")
+	if err != nil {
+		log.Fatalf("failed to read configuration yaml file from ./res folder: %v\n", err)
+	}
+
+	data := make(map[string]any)
+	err = yaml.Unmarshal(contents, &data)
+	if err != nil {
+		log.Fatalf("failed to unmarshal configuration file configuration.yaml: %v\n", err)
+	}
+
+	fmt.Println("data: ", data)
+
+	// convert to struct
+	jsonBytes, err := json.Marshal(data)
+	if err != nil {
+		log.Fatalf("could not marshal map to JSON: %v\n", err)
+	}
+
+	var ovmsClientConf OvmsClientConfig
+	if err := json.Unmarshal(jsonBytes, &ovmsClientConf); err != nil {
+		log.Fatalf("could not unmarshal JSON data to %T: %v", ovmsClientConf, err)
+	}
+
+	fmt.Println("successfully converted to OvmsClientConfig struct", ovmsClientConf)
+
+	//launch the pipeline script from the config
+	if err := launchPipelineScript(ovmsClientConf); err != nil {
+		log.Fatalf("found error while launching pipeline script: %v\n", err)
+	}
+}
+
+func launchPipelineScript(ovmsClientConf OvmsClientConfig) error {
+	scriptDir := "/scripts"
+	scriptFilePath := filepath.Join(scriptDir, ovmsClientConf.OvmsClient.PipelineScript)
+	inputArgs := parseInputArguments(ovmsClientConf)
+	// if running in stream density mode, then the executable is the stream_density shell script itself with input
+	streamDensityMode := os.Getenv("STREAM_DENSITY_MODE")
+	if streamDensityMode == "1" {
+		fmt.Println("in stream density mode!")
+		scriptFilePath = "/home/pipeline-server/stream_density_framework-pipelines.sh"
+		inputArgs = []string{filepath.Join(scriptDir, ovmsClientConf.OvmsClient.PipelineScript)}
+	}
+
+	executable, err := exec.LookPath(scriptFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to get pipeline executable path: %v\n", err)
+	}
+
+	fmt.Println("running executable:", executable)
+	cmd := exec.Command(executable, inputArgs...)
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return fmt.Errorf("failed to get the standard output from executable: %v\n", err)
+	}
+	if err := cmd.Start(); err != nil {
+		return fmt.Errorf("failed to start the pipeline executable: %v\n", err)
+	}
+	reader := bufio.NewReader(stdout)
+	readBytes, _ := reader.ReadString('\n')
+	if err := cmd.Wait(); err != nil {
+		return fmt.Errorf("found error while executing pipeline scripts: %v", err)
+	}
+
+	fmt.Println(string(readBytes))
```

Comment:
instead of using `fmt.Println()`, could we use `log.Println()`. so we can get the timestamps